### ISSUE TITLE
Split simulation cannot be dumped

### ISSUE THREAD

**Issue by Manangka** [Deltares/imod-python#679]
In GitLab by @JoerivanEngelen on Dec 5, 2023, 16:26

To reproduce:

- Split simulation
- Dump splitted simulation

Throws:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
c:\Users\engelen\projects_wdir\alameda_creek\build-model.py in line 3
      187 # %% write model files
----> 189 simulation_split.dump(modeldir_out / "partitioned")
      191 # simulation_split.write(modeldir_out + "/" + name, binary = False)#, validate = False)

File C:\src\imod-python\imod\mf6\simulation.py:699, in Modflow6Simulation.dump(self, directory, validate, mdal_compliant)
    697     else:
    698         path = f"{key}.nc"
--> 699         value.dataset.to_netcdf(directory / path)
    700         toml_content[cls_name][key] = path
    702 with open(directory / f"{self.name}.toml", "wb") as f:

AttributeError: 'list' object has no attribute 'dataset'
```

This is because exchanges are saved in a list. The ``write`` method has this part:

```python
    def write(
...
            elif isinstance(value, list):
                for exchange in value:
                    if isinstance(exchange, imod.mf6.GWFGWF):
                        exchange.write(
                            exchange.packagename(), globaltimes, write_context
                        )
```

We require something similar in the ``dump`` method.

**Comment by JoerivanEngelen** [comment]
Related to https://github.com/Deltares/imod-python/issues/578

**Comment by luitjansl** [comment]
merged in commit 0f0a6e0c45fb05c0537b68e8f7c5fe4ddad73794 into model_splitting_transport_feature
